Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -2544,7 +2660,10 @@ void RenderBlockFlow::computeLogicalLocationForFloat(FloatingObject& floatingObj
         LayoutUnit heightRemainingLeft = 1_lu;
         LayoutUnit heightRemainingRight = 1_lu;
         floatLogicalLeft = logicalLeftOffsetForPositioningFloat(logicalTopOffset, logicalLeftOffset, false, &heightRemainingLeft);
-        while (logicalRightOffsetForPositioningFloat(logicalTopOffset, logicalRightOffset, false, &heightRemainingRight) - floatLogicalLeft < floatLogicalWidth) {
+        auto availableSpace = logicalRightOffsetForPositioningFloat(logicalTopOffset, logicalRightOffset, false, &heightRemainingRight) - floatLogicalLeft;
+        while (availableSpace < floatLogicalWidth) {
+            if (containingBlockMarginTrim.contains(MarginTrimType::InlineStart) && floatLogicalLeft == logicalLeftOffset && availableSpace >= floatWidthWithoutMargin(floatingObject, MarginTrimType::InlineStart))
```

Comment:
Sometimes I find it useful to either paste the relevant parts of the spec or name variables in a way that they are easily recognizable. Something along the lines of
// It also affects floats for which the block container is a containing block by:
// Discarding the inline-start/inline-end margin of an inline-start/inline-end float 
// (or equivalent) whose outer edge on that side coincides with the inner edge of the container.
auto floatHasAdjoiningMargin = containingBlockMarginTrim.contains(MarginTrimType::InlineStart) && floatLogicalLeft == logicalLeftOffset;
if (floatHasAdjoiningMargin && availableSpace >= floatWidthWithoutMargin(floatingObject, MarginTrimType::InlineStart)

Reply:
yeah that is actually really good! Added the helper and added some comments above